src/query/graphite/native/builtin_functions.go

```diff
@@ -242,6 +243,42 @@ func timeShift(
 	}, nil
 }
 
+
+func timeSlice(ctx *common.Context, inputPath singlePathSpec, start string, end string) (ts.SeriesList, error) {
+	tzOffsetForAbsoluteTime := time.Duration(0)
+	startTime, err := graphite.ParseTime(start, time.Now(), tzOffsetForAbsoluteTime)
+	endTime, err := graphite.ParseTime(end, time.Now(), tzOffsetForAbsoluteTime)
+
+	if (err != nil) {
+		return ts.NewSeriesList(), err
+	}
+
+	input := ts.SeriesList(inputPath)
+	output := make([]*ts.Series, input.Len())
+
+	for i, series := range input.Values {
+		nanoSecondsPerStep := series.MillisPerStep() * 1000000
+		stepDuration := time.Duration(nanoSecondsPerStep)
+		truncatedValues := ts.NewValues(ctx, series.MillisPerStep(), series.Len())
+
+		currentTime := series.StartTime()
+		for i := 0; i < series.Len(); i++ {
+			if ( currentTime.After(startTime) && currentTime.Before(endTime)) {
```

Comment:
The graphite documentation says that start and end are inclusive for timeSlice:
https://graphite.readthedocs.io/en/latest/functions.html#graphite.render.functions.timeSlice
>  The start and end times are inclusive.

Hence this should be (also removing the un-needed round brackets):
```go
equalOrAfterStart := currentTime.Equal(startTime) || currentTime.After(startTime)
beforeOrEqualEnd := currentTime.Before(endTime) || currentTime.Equal(endTime)
if equalOrAfterStart && beforeOrEqualEnd {
  // ...
}
```

Reply:
Great catch, thanks